Hierarchical Clustering/Implementation.py

```diff
@@ -0,0 +1,62 @@
+# Mall Customer Segmentation: <font color='blue'>K-Means Clustering</font>
+* Importing the libraries
+* Importing the dataset
+* Dataset information (Pandas Profiling)
+* Elbow method to find the optimal number of clusters
+* Training the K-Means model on the dataset
+* Visualising the clusters
+
+## Importing the libraries
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+"""## Importing the dataset"""
+
+dataset = pd.read_csv('../input/customer-segmentation-tutorial-in-python/Mall_Customers.csv')
+X = dataset.iloc[:, [3, 4]].values
+
+"""## Dataset information (Pandas Profiling)"""
+
+# Commented out IPython magic to ensure Python compatibility.
+import pandas_profiling as pp
+import warnings
+warnings.filterwarnings('ignore')
+# %matplotlib inline
+
+pp.ProfileReport(dataset, title = 'Pandas Profiling report of "dataset"', html = {'style':{'full_width': True}})
+
+"""## Elbow method to find the optimal number of clusters"""
+
+from sklearn.cluster import KMeans
```

Comment:
Since I'm not confident that we will directly use KMeans that's imported,@geekquad can you give it a look